rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,704 @@
+
+As the stream will include potential PII (operator badge information) it MUST be configured as encrypted-at-rest. 
+
+Multiple events can be included in a single Kinesis record, by wrapping them in a JSON array. This is an optimization for improving write speeds: further steps in the Kinesis pipeline may break up or rearrange these arrays.
+
+## Value types
+
+### Car
+Fields:
+- `label` (string | `"none"`, optional): car number corresponding to the GTFS-RT `label` field. Or `"none"` if the inspector has unassigned the car. If the field is absent, it is not modified.
+- `operator` ([Operator](#Operator) | `"none"` | `"unset"`, optional): the operator assigned to this car. The string `"none"` means the inspector has unassigned the operator without reassigning another. The string `"unset"` means to discard the previous value as if the field had never been modified. If the field is absent, it is not modified from its previous value.
+
+`"unset"` and `"none"` are semantically different, and used consistently with the `"unset"` values in [TripUpdated](#TripUpdated). If an operator is `"unset"`, then, as if no update had ever been made, clients SHOULD assume the scheduled operator will operate the car. If an operator is `"none"`, then we don't know who will operate the car.
+
+An empty object `{}` is valid if nothing about the car has been modified.
+
+The length of `cars` always reflects the known length of the train.
+
+### DroppedReason
+An object representing the reason that a trip was dropped. The presence of this object implies that the trip was dropped.
+
+Fields:
+- `reason` (string): free-text description about why a trip was dropped.
+
+Currently the reason is entered by inspectors in a free-text field. It's published in an object so that if Glides collects more structured data in the future, then extra fields can be added. If more fields are added, then a generated text description would be filled into the `reason` field, and it would not be a breaking change.
+
+`reason` MAY be the empty string if for some reason Glides does not have information about why a trip was dropped.
+
+### EditorChange
+Fields in the object:
+- `type` (string `"start"`|`"stop"`): Whether the editor started or stopped editing.
+- `location` ([Location](#Location)): the location the editor started or stopped managing.
+- `editor` ([GlidesUser](#GlidesUser)): the affected editor.
+
+### GlidesUser
+Fields:
+- `emailAddress` (string): the e-mail of the user.
+- `badgeNumber` (string, optional): the badge number of the user. Not all Glides users  have a badge number (for example, CTD contractors).
+
+### Location
+One of:
+- `{"gtfsId": "<GTFS stop ID>"}`, where the GTFS `location_type` is 1 (station).
+- `{"glidesId": "<other>"}`: Other unique identifier for non-revenue locations, internal to Glides.
```

Comment:
I think that could make sense.

So far, we have one for each yard: `yard-reser`, `yard-rside`, `yard-lake`, `yard-ibelt`, `yard-heath`, `yard-matt`. Would all of those have `ops_location_id`s?

We considered using HASTUS places, but chose not to because HASTUS uses separate `ibeltm` and `ibeltu` places and we didn't want to have to distinguish them. Would those have one or two separate location ids?

Can you send us a sample `ops_locations.txt` to the slack #glides channel?

([convenient link to the ODS spec](https://docs.calitp.org/operational-data-standard/spec/))

Reply:
[More discussion in Slack](https://mbta.slack.com/archives/G7UU7PFV4/p1677882894648759?thread_ts=1677863597.074649&cid=G7UU7PFV4)

I've changed `glidesId` to standard `odsId` ids. https://github.com/mbta/technology-docs/pull/18/commits/fc6e5f3d7cf1e831cacf31b24202b5c38f83bec2